service/controller/resource/pvc/local_pvc.go

```diff
@@ -0,0 +1,63 @@
+package pvc
+
+import (
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/provider/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/kvm-operator/service/controller/key"
+)
+
+const (
+	LabelMountTag = "mount-tag"
+)
+
+func newLocalPVCs(customObject v1alpha1.KVMConfig, pvsList *corev1.PersistentVolumeList) []*corev1.PersistentVolumeClaim {
+	var persistentVolumeClaims []*corev1.PersistentVolumeClaim
+
+	for i, workerKVM := range customObject.Spec.KVM.Workers {
+		workerNode := customObject.Spec.Cluster.Workers[i]
+
+		for _, hostVolume := range workerKVM.HostVolumes {
+			for _, pv := range pvsList.Items {
+				if pv.ObjectMeta.Labels[LabelMountTag] == hostVolume.MountTag {
+					persistentVolumeClaim := &corev1.PersistentVolumeClaim{
```

Comment:
This is 4 levels deep. It might be nicer to find the match and then use it instead of using it inside the `for` loop. You can also negate the `if` condition and use a `continue` so the "happy path" is less indented.

Reply:
I made a small refactor. please take a look.

Reply:
Looking better.